raft/src/main/java/org/apache/kafka/raft/LeaderAndEpoch.java

```diff
@@ -28,6 +28,10 @@ public LeaderAndEpoch(OptionalInt leaderId, int epoch) {
         this.epoch = epoch;
     }
 
```

Comment:
Hmm... 

* `Leader#leaderId` seems redundant. Any ID inside the leader object is a leader id by definition.  `Leader#nodeId` expresses what we mean more clearly.

* There should be an accessor function to get the `nodeId` out.

* If we really need a shortcut method for comparing against specific node IDs like this, we could call it `LeaderAndEpoch#hasNodeId(int nodeId)`

Reply:
Also, I just realized that all the data fields in here are public. Can we fix that? In Java we use public accessors, not public data fields, unless there's some really exceptional reason not to.

Reply:
Fixed.